lib/spoom/sorbet/assertions.rb

```diff
@@ -9,150 +9,60 @@ class Assertions
-          Prism::ConstantOperatorWriteNode,
-          Prism::ConstantWriteNode,
-          Prism::ConstantPathAndWriteNode,
-          Prism::ConstantPathOrWriteNode,
-          Prism::ConstantPathOperatorWriteNode,
-          Prism::ConstantPathWriteNode,
-          Prism::GlobalVariableAndWriteNode,
-          Prism::GlobalVariableOrWriteNode,
-          Prism::GlobalVariableOperatorWriteNode,
-          Prism::GlobalVariableWriteNode,
-          Prism::InstanceVariableAndWriteNode,
-          Prism::InstanceVariableOperatorWriteNode,
-          Prism::InstanceVariableOrWriteNode,
-          Prism::InstanceVariableWriteNode,
-          Prism::LocalVariableAndWriteNode,
-          Prism::LocalVariableOperatorWriteNode,
-          Prism::LocalVariableOrWriteNode,
-          Prism::LocalVariableWriteNode,
-        )
       end
 
-      class AssignNode
-        #: AssignType
-        attr_reader :node
+      class Rewriter < Spoom::Visitor
+        ANNOTATION_METHODS = [:let] #: Array[Symbol]
 
-        #: Prism::Location
-        attr_reader :operator_loc
+        LINE_BREAK = "\n".ord #: Integer
+
+        AssignType = T.type_alias do
+          T.any(
+            Prism::ClassVariableAndWriteNode,
+            Prism::ClassVariableOrWriteNode,
+            Prism::ClassVariableOperatorWriteNode,
+            Prism::ClassVariableWriteNode,
+            Prism::ConstantAndWriteNode,
+            Prism::ConstantOrWriteNode,
+            Prism::ConstantOperatorWriteNode,
+            Prism::ConstantWriteNode,
+            Prism::ConstantPathAndWriteNode,
+            Prism::ConstantPathOrWriteNode,
+            Prism::ConstantPathOperatorWriteNode,
+            Prism::ConstantPathWriteNode,
+            Prism::GlobalVariableAndWriteNode,
+            Prism::GlobalVariableOrWriteNode,
+            Prism::GlobalVariableOperatorWriteNode,
+            Prism::GlobalVariableWriteNode,
+            Prism::InstanceVariableAndWriteNode,
+            Prism::InstanceVariableOperatorWriteNode,
+            Prism::InstanceVariableOrWriteNode,
+            Prism::InstanceVariableWriteNode,
+            Prism::LocalVariableAndWriteNode,
+            Prism::LocalVariableOperatorWriteNode,
+            Prism::LocalVariableOrWriteNode,
+            Prism::LocalVariableWriteNode,
+          )
+        end
 
-        #: Prism::Node
-        attr_reader :value, :type
+        #: (String, file: String) -> void
+        def initialize(ruby_contents, file:)
+          super()
 
-        #: (AssignType, Prism::Location, Prism::Node, Prism::Node) -> void
-        def initialize(node, operator_loc, value, type)
-          @node = node
-          @operator_loc = operator_loc
-          @value = value
-          @type = type
+          ruby_contents = ruby_contents.encode("UTF-8") unless @original_encoding == "UTF-8"
+          @node = Spoom.parse_ruby(ruby_contents, file: file) #: Prism::Node
+          @original_encoding = ruby_contents.encoding #: Encoding
```

Comment:
Should this line be above the one where we encode the content as UTF-8? I don't see where `@original_encoding` is set other than here, so I'm thinking that it's `nil` in the check above.

Reply:
Updated in https://github.com/Shopify/spoom/pull/734